implementations/rust/ockam/ockam_abac/src/attribute_access_control.rs

```diff
@@ -94,18 +103,30 @@ impl AbacAccessControl {
     pub async fn is_identity_authorized(&self, id: Identifier) -> Result<bool> {
         let mut environment = self.environment.clone();
 
+        // add the identifier itself as a subject parameter
+        // it's important to do it before we put other attributes, so it can't be overwritten
+        environment.put(
+            format!("{}.{}", SUBJECT_KEY, ABAC_IDENTIFIER_KEY),
+            str(id.to_string()),
+        );
+
         // Get identity attributes and populate the environment:
         match self
             .identities_attributes
             .get_attributes(&id, &self.authority)
             .await?
         {
             Some(attrs) => {
+                environment.put(
+                    format!("{}.{}", SUBJECT_KEY, ABAC_HAS_CREDENTIAL_KEY),
+                    Expr::CONST_TRUE,
+                );
```

Comment:
What does this mean? Is it explained elsewhere how we use this expression?

Reply:
There is a small comment on these constants if you click on them. So, if subject has a valid credential, we'll put `(subject.has_credential, true)` into the environment. So later on it can be checked in ABAC rules

Reply:
Got it, thanks!